test/Microsoft.AspNet.Razor.Runtime.Test/TagHelpers/RuntimeTypeInfoTest.cs

```diff
@@ -0,0 +1,366 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Microsoft.AspNet.Razor.Runtime.TagHelpers
+{
+    public class RuntimeTypeInfoTest
+    {
+        [Theory]
+        [InlineData(typeof(int))]
+        [InlineData(typeof(string))]
+        [InlineData(typeof(Tuple<,>))]
+        [InlineData(typeof(IDictionary<string, string>))]
```

Comment:
Also suggest `Dictionary<string, int>` and something similar to VDD -- `class PartiallyClosedGenericDictionary<TValue> : Dictionary<string, TValue>`